src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivitySourceTests.cs

```diff
@@ -434,6 +434,71 @@ public void TestCreatingActivityUsingDifferentParentIds()
             }).Dispose();
         }
 
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void TestActivityContextIsRemote()
+        {
+            RemoteExecutor.Invoke(() => {
+                ActivityContext ctx = new ActivityContext(ActivityTraceId.CreateRandom(), ActivitySpanId.CreateRandom(), default);
+                Assert.False(ctx.IsRemote);
+
+                bool isRemote = false;
+
+                using ActivitySource aSource = new ActivitySource("RemoteContext");
+                using ActivityListener listener = new ActivityListener();
+                listener.ShouldListenTo = (activitySource) => activitySource.Name == "RemoteContext";
+
+                listener.GetRequestedDataUsingContext = (ref ActivityCreationOptions<ActivityContext> activityOptions) =>
+                {
+                    isRemote = activityOptions.Parent.IsRemote;
+                    return ActivityDataRequest.AllData;
+                };
+
+                ActivitySource.AddActivityListener(listener);
+
+                foreach (bool b in new bool[] { true, false })
+                {
+                    ctx = new ActivityContext(ActivityTraceId.CreateRandom(), ActivitySpanId.CreateRandom(), default, default, b);
+                    Assert.Equal(b, ctx.IsRemote);
+
+                    aSource.StartActivity("a1", default, ctx);
+                    Assert.Equal(b , isRemote);
+                }
+            }).Dispose();
+        }
+
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void TestTraceIdAutoGeneration()
+        {
+            RemoteExecutor.Invoke(() => {
+
+                using ActivitySource aSource = new ActivitySource("TraceIdAutoGeneration");
+                using ActivityListener listener = new ActivityListener();
+                listener.ShouldListenTo = (activitySource) => activitySource.Name == "TraceIdAutoGeneration";
+
+                ActivityContext ctx = default;
+
+                listener.GetRequestedDataUsingContext = (ref ActivityCreationOptions<ActivityContext> activityOptions) =>
+                {
+                    ctx = activityOptions.Parent;
+                    return ActivityDataRequest.AllData;
+                };
+
+                ActivitySource.AddActivityListener(listener);
+
+                aSource.StartActivity("a1", default, ctx);
+                Assert.Equal(default, ctx);
+
+                listener.AutoGenerateRootContextTraceId = true;
+
+                Activity activity = aSource.StartActivity("a2", default, ctx);
+
+                Assert.NotNull(activity);
+                Assert.NotEqual(default, ctx);
+                Assert.Equal(ctx.TraceId, activity.TraceId);
+                Assert.Equal(ctx.SpanId, activity.ParentSpanId);
```

Comment:
what is the value of ctx.SpanId/activity.ParentSpanId? Would they not be null ?

Reply:
If its default, then please validate that they are indeed default.

Reply:
Yes it should be default. I am adding the check
```C#
                Assert.Equal(default(ActivitySpanId).ToHexString(), ctx.SpanId.ToHexString());
```
Note, I am testing the Hex string because the equality depends on if context already created the internal hex string or not.